components/ml/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java

```diff
@@ -2637,4 +2637,35 @@ public void updateSamplePoints(long datasetVersionId, SamplePoints samplePoints)
             MLDatabaseUtils.closeDatabaseResources(connection, updateStatement);
         }
     }
+    
+    @Override
+    public boolean isValidModelStatus(long modelId, int tenantId, String userName) throws DatabaseHandlerException {
+        Connection connection = null;
+        ResultSet result = null;
+        PreparedStatement statement = null;
+        try {
+            connection = dbh.getDataSource().getConnection();
+            statement = connection.prepareStatement(SQLQueries.GET_MODEL_STATUS);
+            statement.setLong(1, modelId);
+            statement.setInt(2, tenantId);
+            statement.setString(3, userName);
+            result = statement.executeQuery();
+            if (result.first()) {
```

Comment:
if we combine it with an AND construct , the exception will be thrown when the model is found but the status in not complete. we need to throw the exception only when the model is not found.